### ISSUE TITLE
[java] UnnecessaryImport false positive for multiline @see Javadoc

### ISSUE THREAD

**Issue by Pankraz76** [pmd/pmd#5832]
**Affects PMD Version:** 7.14.0

**Rule:** [UnnecessaryImport](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryimport)

**Description:**

false positive finding:

```
[WARNING] PMD Failure: io.quarkus.deployment.configuration.BuildTimeConfigBuilderCustomizer:3 Rule:UnnecessaryImport Priority:4 Unused import 'java.util.List'.
[WARNING] PMD Failure: io.quarkus.deployment.configuration.BuildTimeConfigBuilderCustomizer:4 Rule:UnnecessaryImport Priority:4 Unused import 'java.util.Map'.
[WARNING] PMD Failure: io.quarkus.deployment.configuration.BuildTimeConfigBuilderCustomizer:6 Rule:UnnecessaryImport Priority:4 Unused import 'io.quarkus.runtime.LaunchMode'.
```

**Code Sample demonstrating the issue:**

- https://github.com/quarkusio/quarkus/blob/7e64426809af1da2432efea2116a2cf51f0cb425/core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigBuilderCustomizer.java#L23
- https://github.com/quarkusio/quarkus/pull/48414
- https://github.com/quarkusio/quarkus/pull/48414#discussion_r2150880595

```java
package io.quarkus.deployment.configuration;

import java.util.List;
import java.util.Map;

import io.quarkus.runtime.LaunchMode;
import io.quarkus.runtime.console.ConsoleRuntimeConfig;
import io.quarkus.runtime.logging.LogBuildTimeConfig;
import io.quarkus.runtime.logging.LogRuntimeConfig;
import io.quarkus.runtime.logging.LoggingSetupRecorder;
import io.smallrye.config.SmallRyeConfigBuilder;
import io.smallrye.config.SmallRyeConfigBuilderCustomizer;

/**
 * Even if the Log and Console mappings are marked as runtime, they are also used during build time.
 * <p>
 * We cannot register the mappings in the core runtime module because {@link io.smallrye.config.SmallRyeConfig}
 * requires ASM to load the mappings. When we run a Quarkus test, Quarkus will generate the bytecode for the mappings,
 * so we don't need ASM. In a non-Quarkus tests, ASM must be present in the classpath, which we want
 * to avoid (even if they are in the test scope). The logging mappings shouldn't be loaded when running a non-Quarkus
 * test because they are not required.
 *
 * @see LoggingSetupRecorder#initializeBuildTimeLogging(LogRuntimeConfig, LogBuildTimeConfig, ConsoleRuntimeConfig, Map, List,
 *      LaunchMode)
 */
public class BuildTimeConfigBuilderCustomizer implements SmallRyeConfigBuilderCustomizer {
    @Override
    public void configBuilder(final SmallRyeConfigBuilder builder) {
        builder.withMapping(LogBuildTimeConfig.class)
                .withMapping(LogRuntimeConfig.class)
                .withMapping(ConsoleRuntimeConfig.class);
    }
}
```

**Expected outcome:**

no failure.

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI]*

**Comment by oowekyala** [comment]
Please make an effort to fill the issue template properly and add a descriptive title. Also please don't send screenshots, instead copy and paste the relevant text in a code block.

**Comment by Pankraz76** [comment]
updated.

**Comment by pzygielo** [comment]
So, imported types are not used in the code, but in comments only, right?

**Comment by pzygielo** [comment]
Sounds like
- #1639

**Comment by Pankraz76** [comment]
yes.

**Comment by oowekyala** [comment]
Thanks for noticing @pzygielo. I suppose the only difference is that this issue is about `@see` and the other about `{@link}`.

**Comment by Pankraz76** [comment]
thx.
